src/Microsoft.DotNet.ProjectModel/Workspace/ProjectContextEntry.cs

```diff
@@ -0,0 +1,207 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Extensions.ProjectModel;
+using Microsoft.Extensions.ProjectModel.Compilation;
+using Microsoft.Extensions.ProjectModel.Graph;
+
+namespace Microsoft.DotNet.ProjectModel.Workspace
+{
+    internal class ProjectContextEntry
+    {
+        private ProjectContext _context;
+
+        public ProjectContext ProjectContext
+        {
+            get { return _context; }
+            set
+            {
+                if (value == null)
```

Comment:
This is pretty gross. Too much work done in this property. Feels like it should be a method cal or something